src/libguac/guacamole/wol-constants.h

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef __GUAC_WOL_CONSTANTS_H
+#define __GUAC_WOL_CONSTANTS_H
+
+/**
+ * Header file that provides constants and defaults related to libguac
+ * Wake-on-LAN support.
+ * 
+ * @file wol-constants.h
+ */
+
+/**
+ * The default broadcast address to which Wake-on-LAN packets will be sent
+ * if one is not specified.
+ */
+#define GUAC_WOL_DEFAULT_BROADCAST "255.255.255.255"
```

Comment:
Unless there is logic within libguac that uses this value automatically, this doesn't feel like a "default" here (though implementations may well use it as such). According to [RFC 919](https://tools.ietf.org/html/rfc919#section-7), 255.255.255.255 is a special broadcast address for strictly the local network. Perhaps `GUAC_WOL_LOCAL_BROADCAST`, `GUAC_WOL_LOCAL_IPV4_BROADCAST`, or something similar would be better?

Alternatively, providing default logic and using this value if `NULL` is given might make sense. If so, and this remains `GUAC_WOL_DEFAULT_BROADCAST`, I recommend documenting the fact that this is "255.255.255.255", the IPv4 broadcast address for the local network.

Reply:
The logic that uses this by default is in the protocol-specific parts where the setting is pulled in - if there's nothing specified for the broadcast address parameter, this is used.  I think it should remain the default, though I'm happy to name it whatever.  I will update the documentation to reflect that it is the local broadcast address.

Reply:
The reason I have issue with "default" here is that we're essentially _recommending_ that behavior be implemented as a default, not actually implementing a default. It feels like libguac should not concern itself with what protocol implementations may choose to be their defaults.

If some aspect of this WoL implementation implemented its own defaults in the case that specific values are omitted (if `NULL` is passed for the address, a negative value for the timeout, etc.), then I completely agree with documenting these as defaults within libguac since that's exactly what they would be - libguac's defaults. Outside of that, I think the aspect of these values being defaults (or not) would be a private concern of the implementation using libguac.

If the main goal with these defaults is to avoid duplicated code within the bundled protocol support, perhaps some sort of private `defaults.h` within `src/common/` would be better?

Reply:
Okay, I see your point.  I've renamed them, at this point, such that they are not defaults, anymore.  The boot time one is still a little bit of a gray area - I can move that constant definition out to the individual protocols, if you prefer?